src/bun.zig

```diff
@@ -173,7 +173,7 @@ pub const RefCount = @import("./ref_count.zig").RefCount;
 
 pub const MAX_PATH_BYTES: usize = if (Environment.isWasm) 1024 else std.fs.MAX_PATH_BYTES;
 pub const PathBuffer = [MAX_PATH_BYTES]u8;
-pub const WPathBuffer = [MAX_PATH_BYTES / 2]u16;
+pub const WPathBuffer = [windows.PATH_MAX_WIDE]u16;
 pub const OSPathChar = if (Environment.isWindows) u16 else u8;
```

Comment:
☝️ We were over-sizing the path buffer size for windows. The `MAX_PATH_BYTES` is for `u8[]` and is 3x that of the `u16` to account for surrogates etc.
https://github.com/ziglang/zig/blob/f995c1b08ab72897cadbb6a365fb04afb8ea9ca1/lib/std/fs.zig#L43-L47
>
```zig
// Each UTF-16LE character may be expanded to 3 UTF-8 bytes.
// If it would require 4 UTF-8 bytes, then there would be a surrogate
// pair in the UTF-16LE, and we (over)account 3 bytes for it that way.
// +1 for the null byte at the end, which can be encoded in 1 byte.
.windows => os.windows.PATH_MAX_WIDE * 3 + 1,
```

So we can use `std.os.windows.PATH_MAX_WIDE`:
https://github.com/ziglang/zig/blob/f995c1b08ab72897cadbb6a365fb04afb8ea9ca1/lib/std/os/windows.zig#L3699-L3703
>
```zig
/// > The maximum path of 32,767 characters is approximate, because the "\\?\"
/// > prefix may be expanded to a longer string by the system at run time, and
/// > this expansion applies to the total length.
/// from https://docs.microsoft.com/en-us/windows/desktop/FileIO/naming-a-file#maximum-path-length-limitation
pub const PATH_MAX_WIDE = 32767;
```